src/Morphir/IR/NodeId.elm

```diff
@@ -1,34 +1,60 @@
 module Morphir.IR.NodeId exposing (..)
 
 import Morphir.IR.FQName as FQName exposing (FQName)
+import Morphir.IR.Module exposing (ModuleName)
+import Morphir.IR.Name as Name
+import Morphir.IR.Path as Path exposing (Path)
 
 
 type NodeID
     = TypeID FQName
     | ValueID FQName
+    | ModuleID ModuleName
 
 
 nodeIdFromString : String -> Result String NodeID
 nodeIdFromString str =
-    case String.split ":" str of
-        nodeType :: packageName :: moduleName :: localName :: [] ->
-            let
-                fqn : FQName
-                fqn =
-                    FQName.fqn packageName moduleName localName
-            in
-            case nodeType of
+    case String.split ":" str |> List.head of
+        Just prefix ->
+            case prefix of
+                "Value" ->
+                    let
+                        splitNodeString =
+                            String.split ":" str |> List.drop 1
```

Comment:
You are doing the same split twice in this block. The original solution was more elegant and readable. The whole thing can be done in a single pattern match very nicely:

```elm
case String.split ":" str of
    [ "Value", packageName, moduleName, localName ] -> 
        ...
    [ "Type", packageName, moduleName, localName ] -> 
        ...
    [ "Module", packageName, moduleName ] -> 
        ...
    _ ->
        Err <| "Invalid NodeId: " ++ str
```

Reply:
This has been fixed now